activerecord/lib/active_record/encryption/extended_deterministic_queries.rb

```diff
@@ -92,11 +92,11 @@ def exists?(*args)
         end
 
         def find_or_create_by(attributes, &block)
-          find_by(attributes.dup) || create(attributes, &block)
+          find_by(attributes.dup) || create_or_find_by(attributes.dup, &block)
```

Comment:
This `dup` feels like an unfortunate imposition on the caller (here us, but elsewhere potentially user code); can we instead fix `process_encrypted_query_arguments` to switch out the hash for a dup inside the argument list it's working on?

(Not new here, obviously, but I think you've uncovered a behaviour that's worth fixing.)

Reply:
Yes, it's worth exploring, but I'd rather not do it in this PR.

Reply:
Ok, I reverted this part of the diff, and will wait for https://github.com/rails/rails/pull/45722 to be merged first.